packages/client/lib/client/multi-command.ts

```diff
@@ -121,14 +151,23 @@ export default class RedisClientMultiCommand<REPLIES = []> {
   }
 
   static #createScriptCommand(script: RedisScript, resp: RespVersions) {
+    const prefix = scriptArgumentsPrefix(script);
     const transformReply = getTransformReply(script, resp);
+
     return function (this: RedisClientMultiCommand, ...args: Array<unknown>) {
-      this.#multi.addScript(
-        script,
-        script.transformArguments(...args),
+      let redisArgs: CommandArguments;
+
+      const parser = new BasicCommandParser(resp);
+      parser.pushVariadic(prefix);
+      script.parseCommand(parser, ...args);
+
+      redisArgs = parser.redisArgs;
+      redisArgs.preserve = parser.preserve;
+
+      return this.addCommand(
+        redisArgs,
         transformReply
       );
```

Comment:
not using addScript(), probably regression incorrect.  (and similarly for all other users, pool, cluster, sentinel)